### PR TITLE
project reacmadrit

### PR DESCRIPTION
**Pull request by Melinx**

### CODE REVIEW

File: staff/Melinx/js/challenges/scripts/forEach.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+var newArr = [];
```
Comment (Asix94) [inline staff/Melinx/js/challenges/scripts/forEach.js]: Include this array inside the function

File: staff/Melinx/js/challenges/scripts/forEach.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+		// recursive iteration on handler parameter below
+		handler(arr[index], index, arr)
+			newArr.push(arr[index], index, arr);
+		if (++index != arr.length) {
+		arr[0] = "pepe"
+			iterate(index);
+		}
+	};
+	iterate(0);
+	//console.log(newArr)
```
Comment (Asix94) [inline staff/Melinx/js/challenges/scripts/forEach.js]: return newArr

File: staff/Melinx/js/challenges/scripts/map.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+ function map (arr, handler) {
+
+    if (!(arr instanceof Array)) throw Error ('input array is not valid');
+
+    if (!(handler instanceof Function)) throw Error ('handler is not valid');
+    
+    var res = [];
+
+    for (var i = 0; i < array.length; i++) {
```
Comment (Asix94) [inline staff/Melinx/js/challenges/scripts/map.js]: No is array, is arr in your program [arr.length]

File: staff/Melinx/js/challenges/scripts/reduce.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+// return accum
+
+// from Zeke, below:
+
+
+
+
+// from Marina, below:
+
+function reduce(accum, v){
```
Comment (Asix94) [inline staff/Melinx/js/challenges/scripts/reduce.js]: TODO ??

File: staff/Melinx/js/challenges/scripts/toCamelCase.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+'use strict'
+
+var str = "hello My World";
```
Comment (Asix94) [inline staff/Melinx/js/challenges/scripts/toCamelCase.js]: is necesari this variable ??

File: staff/Melinx/js/challenges/scripts/forEach.js (hunk 1 above)
Comment (Melinx) [reply staff/Melinx/js/challenges/scripts/forEach.js]: ok! thx

File: staff/Melinx/js/challenges/scripts/reduce.js (hunk 4 above)
Comment (Melinx) [reply staff/Melinx/js/challenges/scripts/reduce.js]: yes sorry I'll get to it soon.

File: staff/Melinx/js/challenges/vendor/jasmine/3.1.0/spec/PlayerSpec.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
+describe("Player", function() {
```
Comment (manuelbarzi) [inline staff/Melinx/js/challenges/vendor/jasmine/3.1.0/spec/PlayerSpec.js:1]: spec/ folder should not be pushed. they are files just explaining an example in jasmine, but not part of jasmine itself, the library. please, delete spec/ folder, and push again!

File: staff/Melinx/github-app-react-3.0.0/src/App.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+          <Route path="/user/:username" render={props => <Info username={props.match.params.username} showError={this.showError} className="right" />} />
+        </div>
+      </main>
+    )
+  }
+}
+
+// function HelloWorld(props) {
+//   return <div><h1>{props.salute}</h1></div>
+// }
```
Comment (agandia9) [inline staff/Melinx/github-app-react-3.0.0/src/App.js:41]: ....?

File: staff/Melinx/learn-you-react/leanyoureact/src/components/MainContent/MainContent.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import './MainContent.css'
+
+function MainContent(props) {
+
+    if (props.userData.id) {
+
+        return <div className="main">
+            <h2> {`Login: ${props.userData.login}`}</h2>
```
Comment (agandia9) [inline staff/Melinx/learn-you-react/leanyoureact/src/components/MainContent/MainContent.js:9]: props.userData repeat! use destructuring objects in es6

File: staff/Melinx/learn-you-react/leanyoureact/src/components/SearchForm/SearchForm.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import './SearchForm.css'
+
+
+function SeachForm (props){
+    return <form className="form" onSubmit={props.searchUser}>
```
Comment (agandia9) [inline staff/Melinx/learn-you-react/leanyoureact/src/components/SearchForm/SearchForm.js:6]: you can destructuring props..!

**Pull request merged**